weaviate/collection/collection_base.py

```diff
@@ -184,6 +205,35 @@ def update(self, config: CollectionConfigUpdate) -> None:
         if response.status_code != 200:
             raise UnexpectedStatusCodeException("Update collection configuration", response)
 
+    def add_property(self, additional_property: Property) -> None:
```

Comment:
why did you move this? The collection_model has a different mechanism of adding properties (see integration/test_collection_model.py::test_update_properties)

Reply:
It needed to be moved since the adding of a property requires a schema refresh and the schema state is handled within the `_Config` class. But you raise a great point, we can't have `collection_model.config.add_property` and `collection_model.update_model`.

I think the solution is to have this method be private, have `CollectionObject` call it as it is to provide the public `add_property` method to its API, and have `CollectionModelObject` call it within the business logic of its `update_model` function. How does that sound?

Reply:
sounds good!

Reply:
I'm now considering making the compositional attributes of each collection employ inheritance, e.g:
```python
class _ConfigBase:
class _ConfigCollection(_ConfigBase):
class _ConfigCollectionModel(_ConfigBase):

collection.config: _ConfigCollection
collection_model.config: _ConfigCollectionModel
```
which would help to declutter the self referencing between instances up and down the inheritance/composition hierarchy and allow proper hiding of shared private methods.